crates/runtime/src/model/util.rs

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::collections::HashMap;
+
+use llms::bedrock::{BedrockClient, rate_limit::BedrockRateLimitConfigBuilder};
+use secrecy::{ExposeSecret, SecretString};
+use snafu::ResultExt;
+
+/// Extract a secret from a hashmap of secrets, if it exists.
+macro_rules! extract_secret {
+    ($params:expr, $key:expr) => {
+        $params.get($key).map(secrecy::ExposeSecret::expose_secret)
+    };
+}
+
+pub(super) async fn create_bedrock_client(
+    params: &HashMap<String, SecretString>,
+    credential_provider_name: &'static str,
+) -> Result<BedrockClient, Box<dyn std::error::Error + Send + Sync>> {
+    // Build AWS config
+    let mut config_builder = aws_config::defaults(aws_config::BehaviorVersion::latest());
+
+    // Set region if provided
+    if let Some(region) = extract_secret!(params, "aws_region") {
```

Comment:
Note that this won't need to change even if you make the parameter a component parameter (i.e. prefixed)